src/encoder.rs

```diff
@@ -1,3 +1,5 @@
+extern crate alloc;
```

Comment:
If you pursue option 2, I'd prefer to put the extern directive into `lib.rs`

Reply:
Sure, done.

I've been putting it ontop of all modules that use it out of habit personally, to make the requirement explicit. 
(I've only been using Rust for 8-ish months or so, therefore I tread carefully.)

Putting it in `lib.rs` to make it global is not a bad means to do it either.

Reply:
Did you also see my comment above? :sweat_smile: